plugins/search-backend-module-elasticsearch/config.d.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Config {
+  /** Configuration options for the search plugin */
+  search?: {
+    /**
+     * Options for ElasticSearch if selected search engine is ES
+     */
+    elasticSearch?: {
```

Comment:
Would be awesome to encode more of these implicit / dependent config values using discriminating unions and letting the `Config` system do more of the heavy lifting for validation.  Something like...

```typescript
export interface Config {
  search?: {
    elasticSearch?:
      // managed elastic.co
      {
        provider: 'elastic';
        cloudId: string;
        // etc.
      } |
      // managed aws elasticsearch service
      {
        provider: 'aws';
        node: string | string[];
        // etc.
      } |
      // vanilla elasticsearch
      {
        node: string | string[];
        // etc.
      }
  }
}
```

Reply:
Yup, looks better this way to fail startup if the shape is wrong